fedimintd/src/fedimintd.rs

```diff
@@ -249,16 +283,22 @@ async fn run(
     // TODO: Fedimintd should use the config gen API
     // on each run we want to pass the currently passed passsword, so we need to
     // overwrite
-    write_overwrite(opts.data_dir.join(PLAINTEXT_PASSWORD), opts.password)?;
+    if let Some(password) = opts.password {
+        write_overwrite(opts.data_dir.join(PLAINTEXT_PASSWORD), password)?;
+    };
+    let default_params = ConfigGenParamsRequest {
+        meta: BTreeMap::new(),
+        modules: module_gens_params,
+    };
     let mut api = FedimintServer {
         data_dir: opts.data_dir,
         settings: ConfigGenSettings {
-            download_token_limit: cfg.local.download_token_limit,
-            p2p_bind: cfg.local.fed_bind,
-            api_bind: cfg.local.api_bind,
-            p2p_url: cfg.local.p2p_endpoints[&cfg.local.identity].url.clone(),
-            api_url: cfg.consensus.api_endpoints[&cfg.local.identity].url.clone(),
-            default_params: Default::default(),
+            download_token_limit: None,
```

Comment:
What does this configGenSetting do? This seems like a change that may be out of scope for UI stuff.

Reply:
This is how we feed default consensus parameters to the UI. Deployments environments will use this to configure things like networking or which bitcoin node to use. Previously this block of code assumed that configs were already generated, and now it doesn't.